### PR TITLE
Algorithm performance output

### PR DESCRIPTION
**Pull request by jbae11**
This PR simplifies and edits the `algorithm_performance_tests` to:
1. Create a single dictionary variable of all the metrics
2. output text file -> output csv file using pandas dataframe
3. minor code readability edits

### CODE REVIEW

File: d3ploy/solver.py
Comment (katyhuff) [inline d3ploy/solver.py]: why?

File: d3ploy/solver.py
Comment (katyhuff) [inline d3ploy/solver.py]: no blank line needed. PEP8 discourages unnecessary blank lines. https://www.python.org/dev/peps/pep-000
I suggest removing this one as it doesn't add clarity.

File: d3ploy/solver.py
Comment (katyhuff) [inline d3ploy/solver.py]: inline comments aren't needed if the code itself is clear. Since they don't get run or tested, such comments are destined to become out-of-date cruft. 

“Redundant comments are just places to collect lies and misinformation.” 
― Robert C. Martin, Clean Code: A Handbook of Agile Software Craftsmanship

File: tests/unit_tests/deploy_solver_unit_tests.py
Comment (katyhuff) [inline tests/unit_tests/deploy_solver_unit_tests.py:35]: I recommend adding in the docstring that this test is for the constant demand case (if I'm reading this properly).

File: d3ploy/solver.py
Comment (katyhuff) [inline d3ploy/solver.py]: Anyway, this whole section could use some meaningful names. This whole function (beginning https://github.com/ergs/d3ploy/pull/137/files#diff-e29e462b4265bec544e3cd52c7114600R155) seems to be reinventing the OrderedDict. I think this whole function should be deleted and should be replaced by an ordered dict upstream. Up to you, but that would be my recommendation.

**Review by katyhuff — CHANGES_REQUESTED** [review]

**Comment by katyhuff** [comment]
Also, your tests are failing because at some point someone deleted a file (NO_feet_unit_tests.py) and merged a failing PR....

```
=========================== 3 passed in 0.08 seconds ===========================
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-4.0.1, py-1.7.0, pluggy-0.8.0
rootdir: /root/d3ploy, inifile:
plugins: xonsh-0.8.2
collecting ... 
========================= no tests ran in 0.00 seconds =========================
ERROR: file not found: ./tests/integration_unit_tests/NO_fleet_unit_tests.py
```

File: d3ploy/solver.py
Comment (jbae11) [reply d3ploy/solver.py]: you're absolutely right. The entire function was replaced with literally one line..

**Comment by jbae11** [comment]
This PR should maybe wait until the CI fixes are merged in, and pulled

**Comment by gwenchee** [comment]
Since the https://github.com/ergs/d3ploy/pull/140 is stalled, i'm going to merge this PR so that other issues can be worked on

**Pull request merged**
